### PR TITLE
Add runtime _numpy_rc dependency for rc builds and publish to main

### PR DESCRIPTION
**Pull request by h-vetinari**
Modeled after https://github.com/conda-forge/python-feedstock/pull/648, needs https://github.com/conda-forge/staged-recipes/pull/26188; for discussion see https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5790

Created by rebasing #312 onto main (with some selective commit clean-ups) and then applying https://github.com/conda-forge/numpy-feedstock/pull/314/commits/cff4ef3657057215494aa2538cf77b34eaa4db65 + a fresh rerender.

### CODE REVIEW

**Comment by h-vetinari** [comment]
This got discussed in the core call just now, and it's the same as the 3.12rc setup, so I'm going forward with the merge now.

**Comment by jakirkham** [comment]
Thanks Axel! 🙏

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+channel_sources:
+  - conda-forge/label/numpy_rc,conda-forge
```
Comment (jakirkham) [inline recipe/conda_build_config.yaml:2]: It looks like these wound up on `main`

Screenshot 2024-05-01 at 9 13 38 PM

Think we wanted

```suggestion
channel_targets:
- conda-forge numpy_rc
```

Comment (jakirkham) [reply recipe/conda_build_config.yaml:2]: Have manually moved them over to the `numpy_rc` label. Though please recheck

Screenshot 2024-05-01 at 9 16 31 PM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -53,6 +60,9 @@ requirements:
     - liblapack
   run:
     - python
+{% if dev != '' %}
+    - _numpy_rc
+{% endif %}
```
Comment (jakirkham) [inline recipe/meta.yaml:65]: Ah sorry misunderstood this detail ( https://github.com/conda-forge/numpy-feedstock/issues/311#issuecomment-2088915857 ). Now understand the `main` labeled packages are unavailable

Have moved them back

Please double check though

Screenshot 2024-05-01 at 9 22 26 PM

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/conda_build_config.yaml:2]: Sorry I misunderstood. Have reverted this as explained in comment: https://github.com/conda-forge/numpy-feedstock/pull/314/files#r1587043959

Comment (h-vetinari) [reply recipe/conda_build_config.yaml:2]: Indeed, this was fully intentional (as well as documented in various places and referenced in the OP). Please give some time for others to react before you take unilateral action.

I did double check, and all the new numpy builds are on `main`, as they should be.

**Comment by hmaarrfk** [comment]
did this break the Python 3.12 builds?
https://github.com/conda-forge/av-feedstock/pull/71

**Comment by xhochy** [comment]
I'm confused why the numpy pin gets  updated in the PR. It doesn't seem to include any migrations

**Comment by xhochy** [comment]
See https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5790#discussion_r1590007898

**Pull request merged**